exercises/api-authorization/server/authorization-middleware.js

```diff
@@ -0,0 +1,33 @@
+const jwt = require('jsonwebtoken'); // eslint-disable-line
+const ClientError = require('./client-error'); // eslint-disable-line
+
+function authorizationMiddleware(req, res, next) {
+  /* your code here */
+  const token = req.headers['x-access-token'];
+  if (!token) {
+    throw new ClientError(401, 'authentication required');
+  }
+  try {
+    const payload = jwt.verify(token, process.env.TOKEN_SECRET);
+    req.user = payload;
+    next();
+  } catch (err) {
+    console.error(err);
+  }
```

Comment:
If an error happens, you should pass it to `next(err)`. Alternatively, because this function is not asynchronous, you can leave out the `try/catch` block, and Express will send the error to your `errorHandler` for you.